static/app/views/performance/transactionSummary/transactionEvents/eventsTable.tsx

```diff
@@ -372,18 +373,9 @@ class EventsTable extends Component<Props, State> {
     }
 
     const joinCustomData = ({data}: TableData) => {
-      const eventIdMap = {};
-
+      const attachmentsByEvent = groupBy(this.state.attachments, 'event_id');
       data.forEach(event => {
-        event.attachments = [] as any;
-        eventIdMap[event.id] = event;
-      });
-
-      this.state.attachments.forEach(attachment => {
-        const eventAttachments = eventIdMap[attachment.event_id]?.attachments;
-        if (eventAttachments) {
-          eventAttachments.push(attachment);
-        }
+        event.attachments = (attachmentsByEvent[event.id] || []) as any;
```

Comment:
I see why assigning `any` here is necessary, but I don't like it 😔

Reply:
Open to suggestions on this one